rskj-core/src/main/java/co/rsk/peg/LockWhitelist.java

```diff
@@ -33,20 +38,26 @@
  * @author Ariel Mendelzon
  */
 public class LockWhitelist {
-    private List<Address> whitelistedAddresses;
 
-    public LockWhitelist(List<Address> whitelistedAddresses) {
+    private static final Comparator<Address> LEXICOGRAPHICAL_COMPARATOR = (a1, a2)
+            -> UnsignedBytes.lexicographicalComparator().compare(a1.getHash160(), a2.getHash160());
+
+    private SortedMap<Address, Coin> whitelistedAddresses;
+
+    public LockWhitelist(Map<Address, Coin> whitelistedAddresses) {
         // Save a copy so that this can't be modified from the outside
-        this.whitelistedAddresses = new ArrayList<>(whitelistedAddresses);
+        SortedMap<Address, Coin> sortedWhitelistedAddresses = new TreeMap<>(LEXICOGRAPHICAL_COMPARATOR);
```

Comment:
Why are we sorting the entries? If this is for serialization, sorting should be in `BridgeSerializationUtils`, since it's a serialization concern.

Reply:
I'm not sorting here b/c of the serialization, but I'm using this knowledge there for sure. I did it here b/c I wanted to give the `getAddress()` method some _reasonable_ order. In the previous implementation we knew the order would be the insertion order, but now the list is built from [Map#entrySet](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#entrySet--) which doesn't guarantee any order unless the underlying implementation explicitly states it